### PR TITLE
Add `pconstantData` - a direct way of building `Data` encoded constants

### PR DESCRIPTION
**Pull request by TotallyNotChase**
For some reason `ByteString` doesn't have a `ToData` instance so you can't build `PAsData PByteString` directly with `pconstantData`.

You can build newtypes to bytestrings though, stuff like `PAsData PCurrencySymbol`

### CODE REVIEW

**Comment by L-as** [comment]
We could add an instance upstream

**Comment by TotallyNotChase** [comment]
> We could add an instance upstream

I wonder if it should be in `Plutarch.Builtin` - since that's where `pconstantData` is.

**Review by L-as — APPROVED** [review]
LGTM

File: examples/Examples/PIsData.hs [hunk 1]
```diff
@@ -58,9 +55,6 @@ testPDataCompat x =
   PlutusTx.fromData @(PLifted p) (plift $ pforgetData $ pdata $ pconstant @p x)
     @?= Just x
 
-constPAsData :: forall p s. (PlutusTx.ToData (PLifted p)) => PLifted p -> Term s (PAsData p)
-constPAsData x = punsafeCoerce $ pconstant @PData $ PlutusTx.toData x
```
[the remark this answers is not in the document]
Comment (TotallyNotChase) [reply examples/Examples/PIsData.hs:62]: I know right! I discovered this function while adding tests and realized it should actually be in the library

**Pull request merged**